libp2p/dialer.nim

```diff
@@ -174,10 +175,9 @@ proc internalConnect(
   try:
     await lock.acquire()
 
-    if peerId.isSome and reuseConnection:
-      let muxOpt = await self.tryReusingConnection(peerId.get())
-      if muxOpt.isSome:
-        return muxOpt.get()
+    if reuseConnection:
+      self.tryReusingConnection(peerId).withValue(mux):
+        return mux
```

Comment:
if reuseConnection:
      peerIdOpt.map(proc (peerId: PeerId): Opt[Muxer] = self.tryReusingConnection(peerId)).withValue(mux):
          return mux

Reply:
Can we import sugar and do this?
```nim
if reuseConnection:
  peerIdOpt.map((peerId) => self.tryReusingConnection(peerId)).withValue(mux):
      return mux
  ```

Reply:
Tried something else, LMKWYT

Reply:
Looks good.